port/unix/omrsock.c

```diff
@@ -76,8 +76,40 @@ omrsock_getaddrinfo_create_hints(struct OMRPortLibrary *portLibrary, omrsock_add
 
 int32_t
 omrsock_getaddrinfo(struct OMRPortLibrary *portLibrary, char *node, char *service, omrsock_addrinfo_t hints, omrsock_addrinfo_t result)
-{
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+{	
+	omr_os_addrinfo *addrInfoResults = NULL;
+	omr_os_addrinfo *addrInfoHints = NULL;
+	int32_t count = 0;
+	int32_t rc = 0;
+	
+	if (NULL == result) {
+		return OMRPORT_ERROR_INVALID_ARGUMENTS;
+	}
+	
+	if (NULL == hints) {
+		rc = portLibrary->sock_getaddrinfo_create_hints(portLibrary, &hints, 0, 0, 0, 0);
+		if(0 != rc) {
+			return rc;
+		}
+	}
+
+	addrInfoHints = (omr_os_addrinfo *)hints->addrInfo;
+
+	/* Verify that getaddrinfo was successful. */
+	if (0 != getaddrinfo(node, service, addrInfoHints, &addrInfoResults)) {
+		return OMRPORT_ERROR_SOCK_ADDRINFO_FAILED;
+	}
+	memset(result, 0, sizeof(struct OMRAddrInfoNode));
```

Comment:
If the result pointer is valid, I think we should clear it, even (especially?) if there was an error. I think think memset should move to line 88 above.

Reply:
Yes, that makes sense, I will change that.